### PR TITLE
fix: handle invalid contract evm address

### LINKED ISSUE #16949: Handle all exceptions when creating parties for contract transactions
### Description

Certain possible exceptions are not handled when computing the involved parties of contract transactions.  Ensure that all exceptions are handled properly.

### Steps to reproduce

1. See https://production.grafana.hedera-ops.com/goto/K_e94oVNg?orgId=1 for an example transaction 

### Additional context

_No response_

### Hedera network

_No response_

### Version

v0.56

### Operating system

None

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Certain exceptions like invalid evm contract id were not handled in pure checks and when computing parties.
Handle these exceptions correctly.
 
Fixes #16949 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java [hunk 1]
```diff
@@ -370,6 +371,32 @@ void callWhenComputePartiesThrowsException() {
...
+                null);
+        final var context = wellKnownContextWith(blocks, tinybarValues, systemContractGasCalculator);
+        given(worldUpdater.getHederaAccount(SENDER_ID)).willReturn(senderAccount);
+        given(worldUpdater.getHederaAccount(INVALID_CONTRACT_ADDRESS)).willThrow(IllegalArgumentException.class);
+
+        final var handleException = catchThrowableOfType(
+                () -> subject.processTransaction(
+                        transaction, worldUpdater, () -> feesOnlyUpdater, context, tracer, config),
+                HandleException.class);
+        assertThat(handleException.getStatus()).isEqualTo(INVALID_TRANSACTION_BODY);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java:397]: I thought the point of the code at `ContractCallHandler` L91 is that this should return an `INVALID_CONTRACT_ID` not `INVALID_TRANSACTION_BODY`?  Is someone mapping the `PreCheckException` to something else? and losing the more specific error code?

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCallHandlerTest.java [hunk 2]
```diff
@@ -210,4 +214,14 @@ private TransactionBody contractCallTransactionWithInsufficientGas() {
...
                 .build();
     }
+
+    private TransactionBody contractCallTransactionWithInsufficientGas() {
+        final var transactionID = TransactionID.newBuilder().accountID(payer).transactionValidStart(consensusTimestamp);
+        return TransactionBody.newBuilder()
+                .transactionID(transactionID)
+                .contractCall(ContractCallTransactionBody.newBuilder()
+                        .gas(INTRINSIC_GAS_FOR_0_ARG_METHOD - 1)
+                        .contractID(invalidContract))
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCallHandlerTest.java:224]: I think here and L214 are swapped given the method names!

**Review by david-bakin-sl — COMMENTED** [review]
This is what I've got now.  Plus you said elsewhere you'd consider if `EthereumTransactionHandler` (or `EthTxData`) and `ContractCallLocalHandler` need attention.

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCallHandlerTest.java (hunk 2 above)
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCallHandlerTest.java:224]: yep caught that. fixed in update

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java:397]: This is for the more general case if an exception is thrown while computing parties in `TransactionProcessor` that is _not_ already a `HandleException`.  In this case, we may not know what threw the exception but that likely the transaction body was malformed

**Review by kimbor — COMMENTED** [review]
It would be good to have a hapi e2e test for this error condition.

**Comment by lukelee-sl** [comment]
> It would be good to have a hapi e2e test for this error condition.

added

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java [hunk 3]
```diff
@@ -2563,6 +2564,25 @@ final Stream<DynamicTest> lazyCreateInConstructor() {
...
+
+        return hapiTest(
+                withOpContext((spec, ctxLog) -> spec.registry()
+                        .saveContractId(
+                                BAD_EVM_ADDRESS_CONTRACT,
+                                asContractIdWithEvmAddress(ByteString.copyFrom(unhex(BAD_EVM_ADDRESS))))),
+                withOpContext((spec, ctxLog) -> allRunFor(
+                        spec,
+                        contractCallWithFunctionAbi(BAD_EVM_ADDRESS_CONTRACT, getABIFor(FUNCTION, NAME, ERC_721_ABI))
+                                .notTryingAsHexedliteral()
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallSuite.java:2582]: I didn't know this existed!

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Pull request merged**